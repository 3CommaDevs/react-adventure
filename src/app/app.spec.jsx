import React from 'react';
import Enzyme, {shallow} from 'enzyme';

import App from './index.jsx';

describe('<App />', () => {
    it('App component says "App Works!"', () => {
        const wrapper = shallow(<App />);
        const title = wrapper.find('div');
        expect(title.text()).toBe('<HelmetWrapper />App works!');
    });
});
