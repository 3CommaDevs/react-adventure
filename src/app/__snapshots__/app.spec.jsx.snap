// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HelmetWrapper
          defer={true}
          encodeSpecialCharacters={true}
        >
          <title>
            React Redux Starter Pack
          </title>
        </HelmetWrapper>,
        "App works!",
        <br />,
        <button
          onClick={[Function]}
        >
          Async button
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <title>
            React Redux Starter Pack
          </title>,
          "defer": true,
          "encodeSpecialCharacters": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "React Redux Starter Pack",
          },
          "ref": null,
          "rendered": "React Redux Starter Pack",
          "type": "title",
        },
        "type": [Function],
      },
      "App works!",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Async button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Async button",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
          >
            <title>
              React Redux Starter Pack
            </title>
          </HelmetWrapper>,
          "App works!",
          <br />,
          <button
            onClick={[Function]}
          >
            Async button
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <title>
              React Redux Starter Pack
            </title>,
            "defer": true,
            "encodeSpecialCharacters": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "React Redux Starter Pack",
            },
            "ref": null,
            "rendered": "React Redux Starter Pack",
            "type": "title",
          },
          "type": [Function],
        },
        "App works!",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Async button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Async button",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
